kasa/iot/iotdevice.py

```diff
@@ -436,6 +436,16 @@ def update_from_discover_info(self, info: dict[str, Any]) -> None:
             # by the requires_update decorator
             self._set_sys_info(info)
 
+    @staticmethod
+    def extract_sys_info(info: dict[str, Any]) -> dict[str, Any]:
```

Comment:
```suggestion
    def get_sys_info(info: dict[str, Any]) -> dict[str, Any]:
```

Reply:
`get_sys_info` is already a method, albeit that one is `async`. It doesn't seem to cause issues, but should 2 methods have the same name?

Reply:
No, two methods should not have the same name. I think `extract_sys_info` is fine, what do you think, @sdb9696?

Reply:
I think what's bothering me here is that there are two places where there is conditional logic to check for the nested `sys_info`: `get_device_class_from_sys_info` and `extract_sys_info`. I think there should be only one. That's why I suggested a static method on `IotDevice` that is overriden in `IotCamera` to get the nested sysinfo. However from looking through the code paths I think there could be some refactoring to make this work with the discovery `cli`, `_get_device _info` etc. It would be better do this once we have proper test coverage for the kasa cams. For now I've pushed [a commit](https://github.com/python-kasa/python-kasa/pull/1270/commits/05ff1a30095b5f5f958f8417c34a9af925e65f70) to make this a private function so we can change it later.

Reply:
Sounds good to me :+1: